tests/SpecTests/SearchIndexSpecTest.php

```diff
@@ -0,0 +1,218 @@
+     */
+    public function testUpdateSearchIndex(): void
+    {
+        $collection = $this->createCollection($this->getDatabaseName(), $this->getCollectionName());
+        $name = 'test-search-index';
+        $mapping = ['mappings' => ['dynamic' => false]];
+
+        $createdName = $collection->createSearchIndex(
+            $mapping,
+            ['name' => $name],
+        );
+        $this->assertSame($name, $createdName);
+
+        $indexes = $this->waitForIndexes($collection, fn ($indexes) => $this->allIndexesAreQueryable($indexes));
+        $this->assertCount(1, $indexes);
+
+        $mapping = ['mappings' => ['dynamic' => true]];
+        $collection->updateSearchIndex($name, $mapping);
+
+        [$index] = $this->waitForIndexes($collection, fn ($indexes) => $this->allIndexesAreQueryable($indexes));
+
+        $this->assertSame($name, $index->name);
+
+        // Convert to JSON to compare nested associative arrays and nested objects
+        $this->assertJsonStringEqualsJsonString(
+            json_encode($mapping, JSON_THROW_ON_ERROR),
+            json_encode($index->latestDefinition, JSON_THROW_ON_ERROR),
+        );
+    }
+
+    /**
+     * Case 5: dropSearchIndex suppresses namespace not found errors
+     *
+     * @see https://github.com/mongodb/specifications/blob/master/source/index-management/tests/README.rst#case-5-dropsearchindex-suppresses-namespace-not-found-errors
+     */
+    public function testDropSearchIndexSuppressNamespaceNotFoundError(): void
+    {
+        $collection = $this->dropCollection($this->getDatabaseName(), $this->getCollectionName());
+
+        $collection->dropSearchIndex('test-seach-index');
+
+        $this->expectNotToPerformAssertions();
+    }
+
+    /**
+     * Randomize the collection name to avoid duplicate index names when running tests concurrently.
+     * Search index operations are asynchronous and can take up to a few minutes.
+     */
+    protected function getCollectionName(): string
+    {
+        return sprintf('%s.%s', parent::getCollectionName(), bin2hex(random_bytes(5)));
+    }
+
+    private function waitForIndexes(Collection $collection, Closure $callback): array
+    {
+        $timeout = hrtime()[0] + self::WAIT_TIMEOUT_SEC;
+        while (hrtime()[0] < $timeout) {
+            sleep(5);
+            $result = $collection->listSearchIndexes();
+            $this->assertInstanceOf(CachingIterator::class, $result);
+            $result = iterator_to_array($result);
+            if ($callback($result)) {
+                return $result;
+            }
+        }
+
+        $this->fail('Operation did not complete in time');
+    }
+
+    private function allIndexesAreQueryable(array $indexes): bool
+    {
+        if (count($indexes) === 0) {
+            return false;
+        }
+
+        foreach ($indexes as $index) {
+            if (! $index->queryable) {
+                return false;
+            }
+
+            if (! $index->status === 'READY') {
```

Comment:
Should we prefer a constant here instead of a string literal?

Is there any reference for this special value in the documentation/spec?